### PR TITLE
Add ServiceInsight and ServicePulse v2

### LINKED ISSUE #89: Add ServiceInsight and ServicePulse
In the last delivery of the workshop at NDC Oslo 2017, I instructed attendees to look at ServiceInsight as a part of the exercises that they were doing, as well as adding the Saga Audit plugin for the saga exercises.

For the existing NServiceBus users/customers who attended, experiencing Pulse and Insight first-hand was very significant. I heard them saying that there was no way that they'd go back to working without them, almost regardless of how much they cost.

In short, I propose that we incorporate more instruction into the workshop around how to install, configure, and use Pulse and Insight in the workshop.

I'd also suggest that this be prioritized so that it can be done in time for the next delivery of the workshop which, I believe, is Jimmy doing it in NDC Sydney.

### PR DESCRIPTION
**Pull request by dvdstelt**
Fixes #89 

This is an alternate way of adding advanced exercises from https://github.com/Particular/Workshop.Microservices/pull/109
Instead of having module 05 (without any code) the exercise was split into advanced exercises for module 02 and 04.

### CODE REVIEW

File: exercises/03-sagas/instructions.md [hunk 1]
```diff
@@ -232,30 +232,24 @@ In the saga add a new async method called `ProcessOrder` that you will call from
...
 
-This exercise does not require existing NServiceBus skills, but you will probably need to read [the documentation](http://docs.particular.net/nservicebus/sagas/timeouts) to complete it.
+Sagas are excellent for coordinating a business process. In the current saga we're only able to execute the happy path of the business process, where everything works. Payment succeeds and arrives within a short time. But what if the `PaymentSucceededEvent` never arrives?
 
-Sagas are excellent for coordinating a business process. In the current saga we are only able to execute the happy path of the business process. The path where everything goes well. Payment succeeds and arrives within a short time. But what if the `PaymentSucceededEvent` never arrives?
+We can request a timeout in our saga so that, after a certain amount of time when nothing has happened, we can take action such as notifying the shipping or sales department that there may be a problem. A saga in the Finance bounded context would take care of this, but let's also take some action after the timeout elapses. Otherwise, this saga would live forever without anyone noticing the order processing failed in some way.
```
Comment (adamralph) [inline exercises/03-sagas/instructions.md]: There doesn't actually seem to be an exercise here. What do we expect the user to do here?

**Comment by dvdstelt** [comment]
@adamralph Could you have a look at 3.5? I changed it after your comment that there wasn't actually an exercise. I also explain a little bit more about what should go where, and so on. But my English might lacking a bit, so I don't mind if you change it for the better. :-) Thanks!

**Comment by dvdstelt** [comment]
I like it a lot @adamralph ! Thanks! LGTM!

**Comment by adamralph** [comment]
Commits squashed and rebased.

@Particular/microservices-workshop-maintainers any objection to merging.

**Pull request merged**